quarkc/parser.py

```diff
@@ -318,8 +318,8 @@ def visit_types(self, node, (first, rest)):
     def visit_name_re(self, node, children):
         return node.text
 
-    @g.rule('name = _ name_re  _')
-    def visit_name(self, node, (pre, name, post)):
+    @g.rule('name = _ !(keyword !~"[_a-zA-Z0-9]") name_re  _')
+    def visit_name(self, node, (pre, _, name, post)):
         return Name(name)
 
```

Comment:
If we checked for keyword inside visit_name, we could get a nicer error message.